sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -541,6 +553,182 @@ def set_read_only(
+        *,
+        composition_type: Optional[Literal["key", "key_label"]] = None,
+        retention_period: Optional[int] = None,
+        tags: Optional[Dict[str, str]] = None,
+        **kwargs
+    ) -> LROPoller[Snapshot]:
+        """Create a snapshot of the configuration settings.
+
+        :param name: The name of the snapshot to create.
+        :type name: str
+        :param filters: A list of filters used to filter the configuration settings by key field and label field
+            included in the snapshot. Each filter should be in format {"key": <value>, "label": <value>},
+            and key field value is required.
+        :type filters: list[dict[str, str]]
+        :keyword str composition_type: The composition type describes how the key-values
+            within the snapshot are composed. Known values are: "key" and "key_label". The "key" composition type
+            ensures there are no two key-values containing the same key. The 'key_label' composition type ensures
+            there are no two key-values containing the same key and label.
+        :keyword int retention_period: The amount of time, in seconds, that a snapshot will remain in the
+            archived state before expiring. This property is only writable during the creation of a
+            snapshot. If not specified, will set to 2592000(30 days). If specified, should be
+            in range 3600(1 hour) to 7776000(90 days).
+        :keyword dict[str, str] tags: The tags of the snapshot.
+        :return: A poller for create snapshot operation. Call `result()` on this object to wait for the
+            operation to complete and get the created snapshot.
+        :rtype: :class:`~azure.core.polling.LROPoller[~azure.appconfiguration.Snapshot]`
+        """
+        snapshot = Snapshot(
+            filters=filters, composition_type=composition_type, retention_period=retention_period, tags=tags
+        )
+        try:
+            return self._impl.begin_create_snapshot(
+                name=name, entity=snapshot._to_generated(), cls=Snapshot._from_generated, **kwargs
+            )
+        except binascii.Error:
+            raise binascii.Error("Connection string secret has incorrect padding")
+
+    @distributed_trace
+    def archive_snapshot(
+        self,
+        name: str,
+        *,
+        match_condition: MatchConditions = MatchConditions.Unconditionally,
+        etag: Optional[str] = None,
+        **kwargs
+    ) -> Snapshot:
+        """Archive a configuration setting snapshot. It will update the status of a snapshot from "ready" to "archived".
+        The retention period will start to count, the snapshot will expire when the entire retention period elapses.
+
+        :param name: The name of the configuration setting snapshot to archive.
+        :type name: str
+        :keyword match_condition: The match condition to use upon the etag.
+        :type match_condition: :class:`~azure.core.MatchConditions`
```

Comment:
We shouldn't use `:class:` in a `:type:` or `:paramtype:` tag either.
```suggestion
        :type match_condition: ~azure.core.MatchConditions
```